### DISCUSSION TITLE
Question regarding a notifier its family "modifier"

### DISCUSSION THREAD

**Discussion by DJWassink** [rrousselGit/riverpod#2277] [Q&A] [answered]
I have been looking through the docs and am fairly new to Dart / Riverpod but I can't seem to figure out how to pass an argument to a notifier created with a family.

I have created the following provider:
```dart
final entryFormStateProvider =
    AutoDisposeFamilyNotifierProvider<EntryFormStateNotifier, EntryFormState, EntryFormState>(() {
  return EntryFormStateNotifier();
});
```
But I can't figure out how to pass the family argument to the `EntryFormStateNotifier` class. I assume this changed since its not passed to its build method instead of the constructor.
So I assumed Riverpod somehow manages this for me? However I am also confused on how to pass the argument from my widget to the provider, trying the pass the argument like the other providers results in an error:
```dart
ref.watch(entryFormStateProvider(initialEntryFormState));
/// The expression doesn't evaluate to a function, so it can't be invoked.
```

How exactly am I supposed to pass an argument to a NotifierProvider with a family notifier?

Thanks!

**Comment by rrousselGit** [comment, marked as answer]
You don't. The notifier naturally receive the parameter without having to do anything.

Your Notifier base type should be `AutoDisposeFamilyNotifier` (or whatever class with a similar name, I always forget).

**Reply by DJWassink** [reply]
That sounds cool!

However, how do I pass the parameter from my widget to the provider?
```dart
ref.watch(entryFormStateProvider(initialEntryFormState));
```
Is invalid and results in `The expression doesn't evaluate to a function, so it can't be invoked.`

**Reply by rrousselGit** [reply]
Sounds unrelated. Maybe you're missing an import or something

**Reply by DJWassink** [reply]
Ooh I figured it out, I need to type my provider with a `AutoDisposeNotifierProviderFamily` instead of a `AutoDisposeFamilyNotifierProvider`.
Thanks for the help!

**Reply by PackRuble** [reply]
@DJWassink  thank you! I ran into the same problem and still couldn't figure out how to pass the argument! 😅

**Reply by bibek-magar** [reply]
Any updates on how to send params?

**Reply by superakabo** [reply]
@rrousselGit please rename the AutoDisposeFamilyNotifierProvider typedef to prevent confusion.

**Reply by rrousselGit** [reply]
@superakabo Please be more specific. What confusion are you talking about, and why are you using this type? You shouldn't have to type it yourself

**Comment by superakabo** [comment]
> @superakabo Please be more specific. What confusion are you talking about, and why are you using this type? You shouldn't have to type it yourself

I am refactoring my StateNotifierProvider.autoDispose.family implementations to AutoDisposeNotifierProviderFamily. 
While doing that, I mistakenly used AutoDisposeFamilyNotifierProvider which is a typedef instead of AutoDisposeNotifierProviderFamily. The names are quite similar and it is easier to make a mistake when you depend on the intellisense of your IDE.

**Reply by rrousselGit** [reply]
Use Notifier.autoDispose.family, that should be better

**Reply by superakabo** [reply]
OK. Thanks.
